cmd/sonobuoy/app/results.go

```diff
@@ -254,6 +327,75 @@ func printResultsDetails(treePath []string, o *results.Item, input resultsInput)
 	return nil
 }
 
+// printClusterHealthResultsSummary prints the summary of the "fake" plugin for health summary,
+// tryingf to emulate the format of printResultsSummary
+func printClusterHealthResultsSummary(summary discovery.ClusterSummary) error {
+	fmt.Println("Run Details:")
+	fmt.Printf("API Server version: %s\n", summary.APIVersion)
+
+	fmt.Printf("Node health: %d/%d", summary.NodeHealth.Healthy, summary.NodeHealth.Total)
+	//Print the percentage only if Total is not 0 to avoid division by zero errors
+	if summary.NodeHealth.Total != 0 {
+		fmt.Printf(" (%d%%)", 100 * summary.NodeHealth.Healthy / summary.NodeHealth.Total)
+	}
+	fmt.Println()
+	//Details of the failed pods. Checking the slice lenght to avoid trusting the Total
+	if len(summary.NodeHealth.Details) > 0 && summary.NodeHealth.Healthy < summary.NodeHealth.Total {
+		fmt.Println("Details for failed nodes:")
+		for _, node := range summary.NodeHealth.Details {
+			if !node.Healthy {
+				fmt.Printf("%s Ready:%s: %s: %s\n", node.Name, node.Ready, node.Reason, node.Message)
+			}
+		}
+		fmt.Println()
+	}
+
+	//It might be nice to group pods by namespace.
+	//Also here, use len instead of trusting Total
+	if len(summary.PodHealth.Details) > 0 {
+		fmt.Printf("Pods health: %d/%d", summary.PodHealth.Healthy, summary.PodHealth.Total)
+		//Print the percentage only if Total is not 0 to avoid division by zero errors
+		if summary.PodHealth.Total != 0 {
+			fmt.Printf(" (%d%%)", 100 * summary.PodHealth.Healthy / summary.PodHealth.Total)
+		}
+		fmt.Println()
+		if summary.PodHealth.Healthy < summary.PodHealth.Total {
+			pods := make(map[string][]*discovery.HealthInfoDetails)
+			fmt.Println("Details for failed pods:")
+			//First, scan all the pods and extract the not ready ones, and group them by namespace
```

Comment:
We also need to ensure that output is consistent, so we want to sort by namespace/name.

I also think that having the line with `Namespace: %v` is unnecessary sine we're writing it with the pod name.

If it were me I'd sort the slice once by "namespace/name" then iterate over it one time and print the info for the ones that aren't healthy.

That would remove the need for a loop and ensure that the output is perfectly consistent (which is important if we have tests that check the output).